From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Kleinschmidt <jkleinsc@electronjs.org>
Date: Fri, 28 Aug 2026 10:08:46 -0400
Subject: fix: do not pass removed JSPI V8 flag

Chromium's V8 enables JSPI by default and removed the experimental flag,
while Node.js v24.20 passes it unconditionally. Remove the obsolete argument
so Node startup and snapshot generation work with Chromium's V8.

This patch can be removed once Node.js no longer passes the removed flag.

Refs: https://github.com/nodejs/node/pull/59941
Refs: https://chromium-review.googlesource.com/c/v8/v8/+/6937664

diff --git a/src/node.cc b/src/node.cc
index 899c70c790d01d57127461347bae20b0b3a27ed0..3274fafc29c0a8139194272f94db4ec7ea36a861 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -790,11 +790,6 @@ static ExitCode ProcessGlobalArgsInternal(std::vector<std::string>* args,
     // v8_args.emplace_back("--js-source-phase-imports");
   }
 
-#if V8_ENABLE_WEBASSEMBLY
-  // WebAssembly JS Promise Integration
-  v8_args.emplace_back("--experimental-wasm-jspi");
-#endif  // V8_ENABLE_WEBASSEMBLY
-
 #ifdef __POSIX__
   // Block SIGPROF signals when sleeping in epoll_wait/kevent/etc.  Avoids the
   // performance penalty of frequent EINTR wakeups when the profiler is running.
